node/network/src/protocol.rs

```diff
@@ -95,8 +87,8 @@ impl Protocol {
                         }
                     }
                 }
-                Message::BlockResponse(_request_id, blocks) => {
-                    forward_msgs(self.inc_block_tx.clone(), blocks);
+                Message::BlockResponse(_request_id, mut blocks) => {
+                    forward_msgs(self.inc_block_tx.clone(), blocks.drain(..).map(|b| (peer_id, b)).collect());
```

Comment:
Suppose we are at block N, and suddenly we receive announcement of block N+X.
* We send `BlockFetchRequest` for blocks N+1...N+X.
* We receive a response with blocks N+1...N+X and start importing them one by one. First import will trigger `BlockFetchRequest` for blocks N+2...N+X, second for N+3...N+X, etc.
This is already O(X) messages with O(X^2) blocks sent.

We should avoid that and instead change `try_import_blocks` to accept a batch of blocks.

Reply:
Will be fixed by #707 and #708